internal/events/event_poller.go

```diff
@@ -289,20 +313,6 @@ func (ep *eventPoller) waitForShoulderTapOrPollTimeout(lastEventCount int) bool
 		return true
 	}
 
-	// For throughput optimized environments, we can set an eventBatchingTimeout to allow messages to arrive
-	// between polling cycles (at the cost of some dispatch latency)
-	if ep.conf.eventBatchTimeout > 0 && lastEventCount > 0 {
-		shortTimeout := time.NewTimer(ep.conf.eventBatchTimeout)
-		select {
-		case <-shortTimeout.C:
```

Comment:
See in this branch we just fall through to the `longTimeoutDuration` wait, it basically means there was no use of the `eventBatchTimeout` for its stated purpose.

An incomplete implementation had existed previously, and was partially removed leaving this code-cruft.

So in this PR I've come back round and implemented it with a short re-poll cycle (just once) after the batch timeout if the first time we trigger we get an incomplete batch.

It is disabled by default... but you could turn it on with readahead on subscriptions (and for optimization of the event loop), without needing to use batch-based delivery at all.